fastlane_core/lib/fastlane_core/project.rb

```diff
@@ -68,22 +68,13 @@ def select_project(config)
     # Is this project a workspace?
     attr_accessor :is_workspace
 
-    # Should the output of xcodebuild commands be silenced?
-    attr_accessor :xcodebuild_list_silent
-
-    # Should we redirect stderr to /dev/null for xcodebuild commands?
-    # Gets rid of annoying plugin info warnings.
-    attr_accessor :xcodebuild_suppress_stderr
-
     # @param options [FastlaneCore::Configuration|Hash] a set of configuration to run xcodebuild to work out build settings
     # @param xcodebuild_list_silent [Boolean] a flag to silent xcodebuild command's output
     # @param xcodebuild_suppress_stderr [Boolean] a flag to supress output to stderr from xcodebuild
-    def initialize(options, xcodebuild_list_silent: false, xcodebuild_suppress_stderr: false)
```

Comment:
Since this is a blocker for Ruby 3.0 and difficult to fix due to the fact we need to support both `FastlaneCore::Configuration` instance and `Hash` for `options` here, I decided to simply remove them.  For example, Ruby warns below code with this reason "warning: Passing the keyword argument as the last hash parameter is deprecated". It can't achieve both "using default values for xcodebuild_list_silent and xcodebuild_suppress_stderr" and "using keyword arguments syntax for a single positional argument `options`" 😂 

```ruby
FastlaneCore::Project.new(project: 'project')
```

We can come up with a new way to support this capability later without mixing "Hash" and keyword arguments that mess up with Ruby 3.0 if we want🙂 Let me know other fastlane plugins which this change breaks if you have in mind.

Reply:
This is an interesting change 😱 I was not expecting this from Ruby 3.0 but I think I like it 🤔

Reply:
This is kind of inevitable due to this constraint 😅 

> Passing the keyword argument as the last hash parameter is deprecated, or Splitting the last argument into positional and keyword parameters is deprecated